src/state-manager/TransactionConsensus.ts

```diff
@@ -1024,6 +1094,565 @@ class TransactionConsenus {
+            return
+          }
+
+          const executionGroupNodes = new Set(queueEntry.executionGroup.map((node) => node.publicKey))
+          const hasTwoThirdsMajority = this.verifyAppliedReceipt(readableReq, executionGroupNodes)
+          if (!hasTwoThirdsMajority) {
+            /* prettier-ignore */ if (logFlags.error) this.mainLogger.error(`Receipt does not have the required majority for txid: ${readableReq.txid}`)
+            nestedCountersInstance.countEvent('poqo', 'poqo-send-receipt: Rejecting receipt because no majority')
+            return
+          }
+
+          if (logFlags.verbose)
+            this.mainLogger.debug(
+              `POQo: Received receipt from aggregator for ${queueEntry.logID} starting CT2 for data & receipt`
+            )
+          const receivedReceipt = readableReq as AppliedReceipt2
+          queueEntry.poqoReceipt = receivedReceipt
+          queueEntry.appliedReceipt2 = receivedReceipt
+          queueEntry.recievedAppliedReceipt2 = receivedReceipt
+          queueEntry.hasSentFinalReceipt = true
+          Comms.sendGossip(
+            'poqo-receipt-gossip',
+            payload,
+            null,
+            null,
+            queueEntry.transactionGroup,
+            false,
+            4,
+            readableReq.txid,
+            '',
+            true
+          )
+          this.stateManager.transactionQueue.factTellCorrespondingNodesFinalData(queueEntry)
+        } catch (e) {
+          console.error(`Error processing poqoSendReceiptBinary handler: ${e}`)
+          nestedCountersInstance.countEvent('internal', `${route}-exception`)
+          this.mainLogger.error(`${route}: Exception executing request: ${utils.errorToStringFull(e)}`)
+        } finally {
+          profilerInstance.scopedProfileSectionEnd(route)
+        }
+      },
+    }
+
+    Comms.registerInternalBinary(poqoSendReceiptBinary.name, poqoSendReceiptBinary.handler)
+
+    Comms.registerInternal(
+      'poqo-send-vote',
+      async (
+        payload: AppliedVoteHash,
+        _respond: unknown,
+        _sender: unknown,
+        _tracker: string,
+        msgSize: number
+      ) => {
+        profilerInstance.scopedProfileSectionStart('poqo-send-vote', false, msgSize)
+        try {
+          const queueEntry = this.stateManager.transactionQueue.getQueueEntrySafe(payload.txid)
+          if (queueEntry == null) {
+            /* prettier-ignore */ nestedCountersInstance.countEvent('poqo', 'poqo-send-vote: no queue entry found')
+            return
+          }
+          const collectedVoteHash = payload as AppliedVoteHash
+
+          // Check if vote hash has a sign
+          if (!collectedVoteHash.sign) {
+            /* prettier-ignore */ nestedCountersInstance.countEvent('poqo', 'poqo-send-vote: no sign found')
+            return
+          }
+          // We can reuse the same function for POQo
+          this.tryAppendVoteHash(queueEntry, collectedVoteHash)
+        } finally {
```

Comment:
no catch

Reply:
also nuked in other pr

confirmed binary endpoint has the catch